### PR TITLE
Ignore field

### PR DESCRIPTION
**Pull request by tejainece**
Implements feature to request mapper to ignore certain fields.

To ignore certain fields, set the corresponding `view` or `model` in `Field` decorator to empty `String`.

### CODE REVIEW

File: lib/src/mapper_impl.dart [hunk 1]
```diff
@@ -1,50 +1,66 @@
 part of redstone_mapper;
 
-final _defaultFieldDecoder = (Object encodedData, String fieldName, 
-                              Field fieldInfo, List metadata) {
-  var name = fieldInfo.view != null ? fieldInfo.view : fieldName;
-  return (encodedData as Map)[name];
+final _defaultFieldDecoder =
+    (Object encodedData, String fieldName, Field fieldInfo, List metadata) {
+  String lName = fieldName;
```
Comment (Pacane) [inline lib/src/mapper_impl.dart]: what does `lName` mean?

Comment (tejainece) [inline lib/src/mapper_impl.dart]: `Name` is the normal variable name. I use `l` prefix for local variables in a function/method.

Comment (tejainece) [inline lib/src/mapper_impl.dart]: Don't merge this pull request. I have to write a few more test cases. Let me if you like the general idea. I am using a class called `IgnoreValue` to convey from fieldDecoder to ignroe the value.

Comment (Pacane) [inline lib/src/mapper_impl.dart]: ok, well don't do that :P

Also, any reason why you're not using `fieldName` directly? You're not changing its value or do anything bad to it if I read correctly.

File: lib/src/mapper_impl.dart [hunk 2]
```diff
@@ -1,50 +1,66 @@
...
-  if (value != null) {
-    var name = fieldInfo.view != null ? fieldInfo.view : fieldName;
-    encodedData[name] = value;
+final _defaultFieldEncoder = (Map encodedData, String fieldName,
+    Field fieldInfo, List metadata, Object value) {
+  if (value == null) {
+    return;
+  }
+
+  String lName = fieldName;
```
Comment (Pacane) [inline lib/src/mapper_impl.dart]: same here

File: lib/src/mapper_impl.dart (hunk 1 above)
Comment (tejainece) [inline lib/src/mapper_impl.dart]: Don't do prefix? it is easier to differentiate from global and members. People use `k` prefix for constants anyway?

`fieldName` is an argument. I usually have them final, so i don't change it and six months later, assume I haven't changed.

**Comment by Pacane** [comment]
Thanks!

**Pull request merged**